# Red Panda
### A tool for identifying variants in scRNA-seq
#### author: Adam Cornish <adam.cornish@unmc.edu>
#### license: MIT

## Overview

Red Panda employs the unique information found in scRNA-seq to increase accuracy as compared to software designed for bulk sequencing. We utilize the fact that transcripts represented by scRNA-seq reads necessarily only originate from the chromosomes present in a single cell. Where applicable, this fact is used to decide what is and is not a heterozygous variant. For example, if 20% of the transcripts of a gene originate from the maternal chromosome and 80% originate from the paternal, then all the heterozygous variants of that gene in the expressed transcript will be represented at a reference to alternate allele ratio of either 1:4 or 4:1. In other words, all the heterozygous variants in that transcript are expected to be part of a bimodal distribution, which can be exploited to improve the accuracy of variant calling using scRNA-seq data. Such unique information could not be obtained from bulk sequencing, where each variant is independently called. As part of the process of identifying variants, Red Panda creates three different classes: homozygous-looking, bimodally-distributed heterozygous, and non-bimodally-distributed heterozygous.

To run Red Panda, follow these steps:
1. A standard RNA-seq pipeline needs to be run on the sequencing data to generate alignment files and quantification data. We recommend using the bcbio pipeline with STAR or hisat2 as the aligner.
2. Samtools mpileup needs to be run on the alignment files generated by hisat2/STAR to produce a vcf file.
3. Run Red Panda using the quant files from the bcbio run (final/$SAMPLE_NAME/salmon/quant.sf) and the vcf file generated by mpileup
4. Run GATK HaplotypeCaller with --intervals set to a bed file created from this VCF file: data/vcf/to_check/$SAMPLE_NAME.vcf

## Download and Install

    git clone --recursive git://github.com/adambioi/red_panda.git

## Usage
Red Panda requires a number of files to run properly and they are listed below.

    red_panda -q sample_quant.sf -v sample_output.vcf -s sample_name -d base_output_directory -g (hg38|mm10) -A genome_annotation.gtf -V

    -q sample_quant.sf          The expression file created by running sailfish or salmon on the alignment file for your sample.
    -v sample_input.vcf         The VCF file generated by samtools mpileup
    -s sample_name              The sample name to be used in the output files.
    -d base_output_directory    The base output directory. Additional directories will be created within this directory.
    -g (hg38|mm10)              The genome being used; currently only hg38 and mm10 are supported
    -A genome_annotation.gtf    The Ensembl annotation file for the genome being used
    -V                          Verbose debug information

